### PR TITLE
Add documentation for Dev Deployment for 9.1

### PR DESCRIPTION
**Pull request by thiagoelg**

### CODE REVIEW

**Review by LightGuard — CHANGES_REQUESTED** [review]

**Comment by LightGuard** [comment]
Couple of things to clarify or fix, otherwise, good work!

File: tools/create-custom-dev-deployment-image.adoc [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+.Custom Image deployment option
+image::tools/devDeployments/deployment-option-custom-image.png[]
+
+Notice the `Docker Image` parameter; it must be changed to the custom image tag created (`quay.io/myUserName/myDevDeploymentImage:latest` in this example).
+
+Also, notice the `Command` parameter; it will execute `./mvnw quarkus:dev` inside the `$HOME_PATH/app` directory after the upload step is finished. 
+The container's resulting `CMD` will be `dev-deployment-upload-service && ./mvnw quarkus:dev`.
+
+When calling Maven, `./mvnw` is used as the Maven binary is not available globally in this image, and it's not possible to call `mvn` directly.
```
[the remark this answers is not in the document]
Comment (thiagoelg) [reply tools/create-custom-dev-deployment-image.adoc:140]: Hmm.. it is confusing indeed. Let me try a rewrite:

```
Also, notice the `Command` parameter; Its value is the command executed inside the `$HOME_PATH/app` directory after the upload step is finished.
In other words, after the token interpolation is finished, the resulting command executed on the container will be `dev-deployment-upload-service && ./mvnw quarkus:dev`.

This image does not provide the Maven binary globally, so it is not possible to call `mvn` directly. Instead, the image provides a Maven wrapper that can be called with `./mvnw` from the `$HOME_PATH` directory.
```

Is that better?

File: tools/developer-tools-for-vscode.adoc [hunk 2]
```diff
@@ -35,8 +36,9 @@ image::developer-tools/dmn-editor-switch-version.png[]
 
 == Importing DMN Data Types from Java classes
 
-{DEV_TOOLS} allows you to create DMN Data Types directly from Java POJOs. This step-by-step guide aims to 
-describe how effectively using it.
+=== Overview
```
[the remark this answers is not in the document]
Comment (thiagoelg) [reply tools/developer-tools-for-vscode.adoc:39]: This war probably a merge conflict mistake

File: tools/create-custom-dev-deployment-image.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (LightGuard) [reply tools/create-custom-dev-deployment-image.adoc:140]: +1 That works, thank you.

**Review by LightGuard — APPROVED** [review]

**Pull request merged**